Add pr-build-status skill for querying Azure DevOps builds

PureWeen opened the pull request:
<!-- Please let the below note in for people that find this PR -->
> [!NOTE]
> Are you waiting for the changes in this PR to be merged?
> It would be very helpful if you could [test the resulting artifacts](https://github.com/dotnet/maui/wiki/Testing-PR-Builds) from this PR and let us know in a comment if this change resolves your issue. Thank you!

## Description

Adds a new GitHub Copilot CLI skill for querying Azure DevOps build information for PRs.

### What it does

This skill allows Copilot CLI to:
- Get Azure DevOps build IDs associated with a GitHub PR
- Query build status, stages, and failed jobs
- Retrieve detailed UI test results from build logs

### Scripts included

- `Get-PrBuildIds.ps1` - Get build IDs for a PR
- `Get-BuildInfo.ps1` - Get detailed build status
- `Get-TestResults.ps1` - Get test results from build logs

### Security

- All Azure DevOps API calls are to public endpoints (dnceng-public org)
- No authentication tokens or secrets in the code
- GitHub CLI (gh) uses the user's existing authentication

### Example

> can you give me specifics about why iOS UITests Mono  is failing?

<img width="1160" height="706" alt="image" src="https://github.com/user-attachments/assets/220d029b-4fd5-4d76-ab20-595db1a7b105" />

PureWeen committed:
Add pr-build-status skill for querying Azure DevOps builds

kubaflo commented:
@PureWeen can scripts be moved to the scripts folder?😅

```
.github/skills/
└── skill-name/
    ├── SKILL.md          # Required - skill definition
    ├── scripts/          # Optional - executable scripts (self-contained)
    ├── assets/           # Optional - templates/resources
    └── references/       # Optional - documentation
```

kubaflo commented on .github/skills/pr-build-status/SKILL.md:
```diff
@@ -0,0 +1,51 @@
+---
+name: pr-build-status
+description: "Retrieve Azure DevOps build information for GitHub Pull Requests, including build IDs, stage status, and failed jobs."
+---
```
To keep it consistent with our skills:
```
metadata:
  author: dotnet-maui
  version: "1.0"
compatibility: Requires GitHub CLI (gh) authenticated with access to dotnet/maui repository.
```

PureWeen commented:
/azp run

PureWeen committed:
Address PR review feedback and improve build error detection

- Add metadata (author, version, compatibility) to SKILL.md
- Fix bash/pwsh documentation inconsistency
- Move scripts to scripts/ subfolder for consistency
- Fix PowerShell variable escaping (${BuildId} syntax)
- Fix deduplication sort in Get-PrBuildIds.ps1
- Rename Get-TestResults.ps1 to Get-BuildErrors.ps1
- Rewrite script to detect both build errors and test failures
- Add -ErrorsOnly and -TestsOnly switches
- Search failed Tasks (not just Jobs) for MSBuild errors
